vllm/v1/spec_decode/eagle.py

```diff
@@ -24,6 +24,34 @@ def __init__(
         self.arange = torch.arange(vllm_config.scheduler_config.max_num_seqs,
                                    device=device)
 
+        max_batch_size = vllm_config.scheduler_config.max_num_seqs
+        vocab_size = vllm_config.model_config.get_vocab_size()
+
+        # setup buffers for draft token ids and probs to be
+        # reused across steps for Rejection Sampling
+        self.curr_batch_size = -1
+        self._draft_token_ids_buffer = torch.zeros(max_batch_size,
+                                                   self.num_speculative_tokens,
+                                                   dtype=torch.int32,
+                                                   device=device)
+        self._draft_token_ids_buffer_shape = self._draft_token_ids_buffer.shape
+        self._draft_probs_buffer = torch.zeros(max_batch_size,
+                                               self.num_speculative_tokens,
+                                               vocab_size,
+                                               dtype=torch.float32,
+                                               device=device)
+        self._draft_probs_buffer_shape = self._draft_probs_buffer.shape
```

Comment:
Just want to point this out, we might need to fix it. For the `draft_probs_buffer`, it has size (plug in numbers of llama3-8B):
256 * 10 * 128256 * 4 / 1024 / 1024 = 1.3G
It has a low probability that this might trigger OOM if we do this after vLLM preallocates all memory for kv cache. But it should not be a big problem.

Reply:
Is there a way to allocate this before vLLM preallocates memory for KVC?

Reply:
The preallocation of RS buffer is happening [here](https://github.com/vllm-project/vllm/blob/cd77382ac176b3acc717d79be7981dc60e71dbe3/vllm/v1/worker/gpu_model_runner.py#L161-L173) which is when gpuModelRunner is created. Could you point me to which line of code computes the available GPU memory and allocated the KVC on that?